From 48ff16650348b44ae3d78d7cc7b2dc2d92309de7 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Caol=C3=A1n=20McNamara?= <caolanm@redhat.com>
Date: Thu, 17 Jul 2014 13:48:30 +0100
Subject: [PATCH] Related: rhbz#1117853 nStartPara of EE_PARA_NOT_FOUND

we know that nStartPara is EE_PARA_NOT_FOUND so rOutliner.GetAbsPos must have
returned that, but we don't know the circumstances that lead to that yet.

Change-Id: I53a488317d154b4a3c050248b8737da0a611ca43
---
 editeng/source/outliner/outliner.cxx | 2 +-
 sd/source/ui/view/outlnvsh.cxx       | 3 +--
 2 files changed, 2 insertions(+), 3 deletions(-)

diff --git a/editeng/source/outliner/outliner.cxx b/editeng/source/outliner/outliner.cxx
index a8e3d66..6aaa3fe 100644
--- a/editeng/source/outliner/outliner.cxx
+++ b/editeng/source/outliner/outliner.cxx
@@ -399,7 +399,7 @@ OutlinerParaObject* Outliner::CreateParaObject( sal_Int32 nStartPara, sal_Int32
     if ( ( nStartPara + nCount ) > pEditEngine->GetParagraphCount() )
         nCount = pEditEngine->GetParagraphCount() - nStartPara;
 
-    if( !nCount )
+    if (nCount <= 0)
         return NULL;
 
     EditTextObject* pText = pEditEngine->CreateTextObject( nStartPara, nCount );
diff --git a/sd/source/ui/view/outlnvsh.cxx b/sd/source/ui/view/outlnvsh.cxx
index f787857..8ff50d1 100644
--- a/sd/source/ui/view/outlnvsh.cxx
+++ b/sd/source/ui/view/outlnvsh.cxx
@@ -1674,7 +1674,6 @@ bool OutlineViewShell::UpdateTitleObject( SdPage* pPage, Paragraph* pPara )
 
     ::Outliner*             pOutliner = pOlView->GetOutliner();
     SdrTextObj*         pTO  = pOlView->GetTitleTextObject( pPage );
-    OutlinerParaObject* pOPO = NULL;
 
     OUString aTest = pOutliner->GetText(pPara);
     bool    bText = !aTest.isEmpty();
@@ -1691,9 +1690,9 @@ bool OutlineViewShell::UpdateTitleObject( SdPage* pPage, Paragraph* pPara )
         }
 
         // if we have a title object and a text, set the text
+        OutlinerParaObject* pOPO = pTO ? pOutliner->CreateParaObject(pOutliner->GetAbsPos(pPara), 1) : NULL;
         if( pTO )
         {
-            pOPO = pOutliner->CreateParaObject( pOutliner->GetAbsPos( pPara ), 1 );
             pOPO->SetOutlinerMode( OUTLINERMODE_TITLEOBJECT );
             pOPO->SetVertical( pTO->IsVerticalWriting() );
             if( pTO->GetOutlinerParaObject() && (pOPO->GetTextObject() == pTO->GetOutlinerParaObject()->GetTextObject()) )
-- 
1.9.3

